### PR TITLE
User Avatar service response format change

### PR DESCRIPTION
**Pull request by pchojnacki**
- remove RepresentationFactory that obscured response format
- store whole image URLs in Attribute service instead of image id

Stored URLs point to configured Vignette thumbnailer instance.
/cc: @sqreek

### CODE REVIEW

File: service/user-avatar/src/main/java/com/wikia/services/useravatar/UserAvatarApplication.java [hunk 1]
```diff
@@ -18,10 +18,6 @@
 import org.slf4j.LoggerFactory;
 
 public class UserAvatarApplication extends ConsulGuiceApplication<UserAvatarConfiguration> {
-
-  public static final RepresentationFactory
-      REPRESENTATION_FACTORY = new StandardRepresentationFactory();
-
   public static final Logger LOGGER = LoggerFactory.getLogger(UserAvatarApplication.class);
```
Comment (WikiaTech) [inline service/user-avatar/src/main/java/com/wikia/services/useravatar/UserAvatarApplication.java]: MAJOR Make the "LOGGER" logger private static final. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1312)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 7 issues:
* MAJOR 5 major
* MINOR 2 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MINOR [Remove this unused import 'com.theoryinpractise.halbuilder.standard.StandardRepresentationFactory'.](https://github.com/Wikia/pandora/blob/c0d66059ef522e58ccfd9ed82b57728b73f384df/service/user-avatar/src/main/java/com/wikia/services/useravatar/UserAvatarApplication.java#L13) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AUselessImportCheck)
* MINOR [Remove this unused import 'com.theoryinpractise.halbuilder.api.RepresentationFactory'.](https://github.com/Wikia/pandora/blob/c0d66059ef522e58ccfd9ed82b57728b73f384df/service/user-avatar/src/main/java/com/wikia/services/useravatar/UserAvatarApplication.java#L11) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AUselessImportCheck)
* MAJOR [Define and throw a dedicated exception instead of using a generic one.](https://github.com/Wikia/pandora/blob/c0d66059ef522e58ccfd9ed82b57728b73f384df/service/user-avatar/src/main/java/com/wikia/services/useravatar/guice/UserAvatarModule.java#L59) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00112)
* MAJOR [Refactor this method to throw at most one checked exception instead of: java.io.IOException, java.text.ParseException](https://github.com/Wikia/pandora/blob/c0d66059ef522e58ccfd9ed82b57728b73f384df/service/user-avatar/src/main/java/com/wikia/services/useravatar/resources/UserAvatarResource.java#L135) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1160)
* MAJOR [Refactor this method to throw at most one checked exception instead of: java.io.IOException, java.text.ParseException](https://github.com/Wikia/pandora/blob/c0d66059ef522e58ccfd9ed82b57728b73f384df/service/user-avatar/src/main/java/com/wikia/services/useravatar/service/ImageService.java#L90) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1160)
* MAJOR [Move this closing curly brace to the next line.](https://github.com/Wikia/pandora/blob/c0d66059ef522e58ccfd9ed82b57728b73f384df/service/user-avatar/src/test/java/com/wikia/services/useravatar/resources/UserAvatarResourceTest.java#L198) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ARightCurlyBraceStartLineCheck)

File: service/user-avatar/src/main/java/com/wikia/services/useravatar/UserAvatarApplication.java [hunk 2]
```diff
@@ -18,12 +18,6 @@
 import org.slf4j.LoggerFactory;
```
Comment (WikiaTech) [inline service/user-avatar/src/main/java/com/wikia/services/useravatar/UserAvatarApplication.java]: MINOR Remove this unused import 'org.slf4j.LoggerFactory'. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AUselessImportCheck)

File: service/user-avatar/src/main/java/com/wikia/services/useravatar/resources/UserAvatarResource.java [hunk 3]
```diff
@@ -1,5 +1,7 @@
 package com.wikia.services.useravatar.resources;
 
+import static javax.ws.rs.core.MediaType.APPLICATION_JSON;
+
 import com.wikia.dropwizard.auth.AuthenticatedUser;
 import com.wikia.pandora.http.StandardResponses;
 import com.wikia.services.useravatar.UserAvatarApplication;
```
Comment (WikiaTech) [inline service/user-avatar/src/main/java/com/wikia/services/useravatar/resources/UserAvatarResource.java]: MINOR Remove this unused import 'com.wikia.services.useravatar.UserAvatarApplication'. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues:
* MINOR 3 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MINOR [Remove this unused import 'org.slf4j.Logger'.](https://github.com/Wikia/pandora/blob/06e6d9e1f3f91a07e5aaf5aed6ff7b19c0a70b4f/service/user-avatar/src/main/java/com/wikia/services/useravatar/UserAvatarApplication.java#L17) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AUselessImportCheck)

File: service/user-avatar/src/main/java/com/wikia/services/useravatar/resources/UserAvatarResource.java [hunk 4]
```diff
@@ -45,37 +44,32 @@
...
     this.imageService = imageService;
-    this.userAvatarAttributeImageIdPrefix = userAvatarAttributeImageIdPrefix;
-  }
-
-  public URI createUserAvatarSelfUrl(Integer userId) {
-    return UriBuilder.fromResource(UserAvatarResource.class).build(userId);
+    this.imageUrlBuilder = imageUrlBuilder;
   }
 
   private void ensureAuthUserIdIsSameAsUserId(Integer userId, AuthenticatedUser user) {
```
Comment (sqreek) [inline service/user-avatar/src/main/java/com/wikia/services/useravatar/resources/UserAvatarResource.java]: Consider extract this to library.
It's at least third place where we use same code (look at UserPreference service)

**Comment by sqreek** [comment]
LGTM

**Pull request merged**